code/game/objects/effects/effect_system.dm

```diff
@@ -488,11 +472,18 @@ steam.start() -- spawns the effect
 				msg_admin_attack("A chemical smoke reaction has taken place in ([whereLink]). No associated key.", 0, 1)
 				log_game("A chemical smoke reaction has taken place in ([where])[contained]. No associated key.")
 
-	start()
+	start(effect_range = 2)
 		var/i = 0
 
 		var/color = mix_color_from_reagents(chemholder.reagents.reagent_list)
-
+		var/obj/effect/effect/chem_smoke/smokeholder = new /obj/effect/effect/chem_smoke(src.location)
+		for(var/atom/A in view(effect_range, smokeholder))
+			chemholder.reagents.reaction(A)
+			if(iscarbon(A))
+				var/mob/living/carbon/C = A
+				if(!(C.wear_mask && (C.internals != null || C.wear_mask.flags & BLOCK_GAS_SMOKE_EFFECT)))
+					chemholder.reagents.copy_to(C, chemholder.reagents.total_volume)

```

Comment:
There is a slight problem with that, it will mean infinitely looping smoke would be possible. I'd honestly prefer to have it reduce the amount for each transfer.